n3fit/src/n3fit/layers/msr_normalization.py

```diff
@@ -27,19 +33,23 @@ def __init__(self, output_dim=14, mode="ALL", **kwargs):
         else:
             raise ValueError(f"Mode {mode} not accepted for sum rules")
 
-        idx = []
+        self.indices = []
         if self._msr_enabled:
-            idx += GLUON_IDX
+            self.indices += [IDX['g']]
         if self._vsr_enabled:
-            idx += V_IDX + V3_IDX + V8_IDX + V15_IDX
-
-        self._out_scatter = op.as_layer(
-            op.scatter_to_one, op_kwargs={"indices": idx, "output_dim": output_dim}
-        )
+            self.indices += [IDX[f] for f in ['v', 'v35', 'v24', 'v3', 'v8', 'v15']]
+        self.indices = [[i] for i in self.indices]
 
         super().__init__(**kwargs)
 
-    def call(self, pdf_integrated, photon_integral):
+    def build(self, input_shape):
+        out_shape = input_shape[1:]
+        self._out_scatter = lambda pdf_integrated: op.scatter_to_one(
+            pdf_integrated, indices=self.indices, output_shape=out_shape
+        )
+        super().build(input_shape)
```

Comment:
Since you moved this to `build`, I'd suggest moving the indices part also here, that way you avoid filling up a `self.indices` variable in the middle.

Also, is there a reason you prefer the `lambda`? Because you could also avoid the call to `build` and just do `scatter_to_one` later (I mean, once you are changing the entire layer there's no reason to constrain yourself to the previous one as long as input and output are the same)

Reply:
Good point, moved it to call.

I actually had a bigger rewrite of this layer, removing all of the indexing in the call and replacing it with tensor operations. But I had to do so many tricks to get the broadcasting to work correctly that in the end it was harder to read than what there is now, so I discarded it.